sysroot/usr/include/kernel/lib/mutex.hpp

```diff
@@ -14,6 +14,9 @@
 
 typedef struct px_mutex {
     bool locked;
+    px_mutex() {
+        locked = false;
+    };
```

Comment:
Also as an implementation thing, I think we should define as little as possible in header files.

The constructor declaration should be in the header and the implementation in the cpp file.

Reply:
I honestly have no clue how to do that with structs. They're basically classes, so should I just declare `px_mutex();` in the header and move everything else, struct and all into the implementation file?

Reply:
The only difference in C++ between structs and classes is this:
- `class` members are `private` by default
- `struct` members are `public` by default
They are exactly the same otherwise.

Reply:
The resulting header code would look like this
```c++
typedef struct px_mutex {
    bool locked;
    px_mutex();
} px_mutex_t;
```

Reply:
Okay, in the header I have
```C++
typedef struct px_mutex {
    bool locked;
    px_mutex();
} px_mutex_t;
```
and in the source file I have
```C++
px_mutex::px_mutex() {
    locked = false;
};
```
Does that seem right?

Reply:
> Okay, in the header I have
> 
> ```c++
> typedef struct px_mutex {
>     bool locked;
>     px_mutex();
> } px_mutex_t;
> ```
> 
> and in the source file I have
> 
> ```c++
> px_mutex::px_mutex() {
>     locked = false;
> };
> ```
> 
> Does that seem right?

Perfect!

Reply:
I haven't committed that yet so I'll go ahead and do that for both the mutexes and semaphores.

Reply:
Changed in cd4204d